v2/kafka-to-pubsub/README.md

```diff
@@ -0,0 +1,147 @@
+
+```
+IMAGE_NAME=<my-image-name>
+TARGET_GCR_IMAGE=gcr.io/${PROJECT}/${IMAGE_NAME}
+BASE_CONTAINER_IMAGE=<my-base-container-image>
+BASE_CONTAINER_IMAGE_VERSION=<my-base-container-image-version>
+TEMPLATE_PATH="gs://${BUCKET_NAME}/templates/kafka-pubsub.json"
+TARGET_GCR_IMAGE=gcr.io/${PROJECT}/${IMAGE_NAME}
+```
+
+## Build Apache Kafka to Google Pub/Sub Flex Dataflow Template
+
+Dataflow Flex Templates package the pipeline as a Docker image and stage these images 
+on your project's [Container Registry](https://cloud.google.com/container-registry).
+
+### Assembling the Uber-JAR
+
+The Dataflow Flex Templates require your Java project to be built into 
+an Uber JAR file.
+
+Navigate to the v2 folder:
+
+```
+cd /path/to/DataflowTemplates/v2
+```
+
+Build the Uber JAR:
+
+```
+mvn package -am -pl kafka-to-pubsub
+```
+
+ℹ️ An **Uber JAR** - also known as **fat JAR** - is a single JAR file that contains 
+both target package *and* all its dependencies.
+
+The result of the `package` task execution is a `kafka-to-pubsub-1.0-SNAPSHOT.jar` 
+file that is generated under the `target` folder in kafka-to-pubsub directory.
+
+### Creating the Dataflow Flex Template
+
+To execute the template you need to create the template spec file containing all
+the necessary information to run the job. This template already has the following [metadata
+file](kafka-to-pubsub/src/main/resources/kafka_to_pubsub_metadata.json) in resources.
+
+Navigate to the template folder:
+
+```
+cd /path/to/DataflowTemplates/v2/kafka-to-pubsub
+```
+
+Build the Dataflow Flex Template:
+
+```
+gcloud dataflow flex-template build ${TEMPLATE_PATH} \
+       --image-gcr-path "{$TARGET_GCR_IMAGE}" \
+       --sdk-language "JAVA" \
+       --flex-template-base-image ${BASE_CONTAINER_IMAGE} \
+       --metadata-file "src/main/resources/kafka_to_pubsub_metadata.json" \
+       --jar "target/kafka-to-pubsub-1.0-SNAPSHOT.jar" \
+       --env FLEX_TEMPLATE_JAVA_MAIN_CLASS="com.google.cloud.teleport.v2.KafkaToPubsub"
+```
+
+### Create Dataflow Job Using the Apache Kafka to Google Pub/Sub Dataflow Flex Template
+
+To deploy the pipeline, you should refer to the template file and pass the 
+[parameters](https://cloud.google.com/dataflow/docs/guides/specifying-exec-params#setting-other-cloud-dataflow-pipeline-options) 
+required by the pipeline.
+
+You can do this in 3 different ways:
+1. Using [Dataflow Google Cloud Console](https://console.cloud.google.com/dataflow/jobs)
+
+2. Using `gcloud` CLI tool
+    ```
+    gcloud dataflow flex-template run "kafka-to-pubsub-`date +%Y%m%d-%H%M%S`" \
+        --template-file-gcs-location "${TEMPLATE_PATH}" \
+        --parameters bootstrapServers="broker_1:9092,broker_2:9092" \
+        --parameters inputTopics="topic1,topic2" \
+        --parameters outputTopic="projects/${PROJECT}/topics/your-topic-name" \
+        --region "${REGION}"
+    ```
+3. With a REST API request
+    ```
+    API_ROOT_URL="https://dataflow.googleapis.com"
```

Comment:
Flex templated has new API EndPoint https://dataflow.googleapis.com/v1b3/projects/$PROJECT/locations/us-central1/flexTemplates:launch.
Refer [docs](https://cloud.google.com/dataflow/docs/guides/templates/using-flex-templates#running_a_flex_template_pipeline) for REST API request

Reply:
fixed